pennylane/workflow/set_shots.py

```diff
@@ -12,25 +12,20 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """
-Contains the set_shots transform, which sets the number of shots for a given tape.
+This module contains the set_shots decorator.
 """
-from typing import Sequence, Union
+from typing import Sequence, Tuple, Union
 
 from pennylane.measurements import Shots
-from pennylane.tape import QuantumScript, QuantumScriptBatch
-from pennylane.transforms.core import transform
-from pennylane.typing import PostprocessingFn
 
+from .qnode import QNode
 
-def null_postprocessing(results):
-    """An empty post-processing function."""
-    return results[0]
 
-
-@transform
 def set_shots(
-    tape: QuantumScript, shots: Union[Shots, None, int, Sequence[Union[int, tuple[int, int]]]]
-) -> tuple[QuantumScriptBatch, PostprocessingFn]:
+    qnode: QNode,
+    *,
```

Comment:
```suggestion
```
Since there's only two arguments, I don't think there's really that much ambiguity in the argument order. I think it's fine to allow `set_shots(qnode, 10)`.